.github/workflows/tests.yml

```diff
@@ -0,0 +1,352 @@
+    paths-ignore:
+      - 'docs/**'
+
+jobs:
+  lint:
+    name: 'lint (node: ${{ matrix.node }})'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [12, 14]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Install modules
+        run: yarn
+      - name: Run lint
+        run: yarn run -s lint
+
+  unit_back:
+    name: 'unit_back (node: ${{ matrix.node }})'
+    needs: [lint]
+    runs-on: ubuntu-latest
+    env:
+      CODECOV_TOKEN: ${{ secrets.codecov }}
+    strategy:
+      matrix:
+        node: [12, 14]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Install modules
+        run: |
+          yarn && yarn global add codecov
+          echo "$(yarn global bin)" >> $GITHUB_PATH
+      - name: Run test
+        run: yarn run -s test:unit --coverage && codecov -C -F unit
+
+  unit_front:
+    name: 'unit_front (node: ${{ matrix.node }})'
+    needs: [lint]
+    runs-on: ubuntu-latest
+    env:
+      CODECOV_TOKEN: ${{ secrets.codecov }}
+    strategy:
+      matrix:
+        node: [12, 14]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Install modules
+        run: |
+          yarn && yarn global add codecov
+          echo "$(yarn global bin)" >> $GITHUB_PATH
+      - name: Run test
+        run: yarn run -s test:unit && codecov -C -F front
+
+  e2e_ce_pg:
+    runs-on: ubuntu-latest
+    needs: [lint, unit_back, unit_front]
+    name: 'E2E (pg:${{ matrix.postgres }} node: ${{ matrix.node }})'
+    strategy:
+      matrix:
+        node: [12, 14]
+        postgres: [12]
+      max-parallel: 2
+    services:
+      postgres:
+        # Docker Hub image
+        image: postgres:${{ matrix.postgres }}
+        # Provide the password for postgres
+        env:
+          POSTGRES_USER: strapi
+          POSTGRES_PASSWORD: strapi
+          POSTGRES_DB: strapi_test
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432:5432
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Install modules
+        run: |
+          yarn && yarn global add wait-on
+          echo "$(yarn global bin)" >> $GITHUB_PATH
+      - name: Generate test app
+        run: STRAPI_DISABLE_EE=true yarn run -s test:generate-app --dbclient=postgres --dbhost=localhost --dbport=5432 --dbname=strapi_test --dbusername=strapi --dbpassword=strapi
```

Comment:
To avoid duplicating the credentials/db names, you could use a job-scoped env and reuse the variables both in the postgres's service and here (same for other DBs). What do you think?

Reply:
this won't work as the job env is not shared that way